index/scorch/segment/zap/contentcoder.go

```diff
@@ -34,10 +34,14 @@ var termSeparator byte = 0xff
 var termSeparatorSplitSlice = []byte{termSeparator}
 
 type chunkedContentCoder struct {
-	final        []byte
-	chunkSize    uint64
-	currChunk    uint64
-	chunkLens    []uint64
+	final     []byte
+	chunkSize uint64
+	currChunk uint64
+	chunkLens []uint64
+
+	w                io.Writer
+	writeOutInChunks bool
```

Comment:
minor naming thing -- i have the feeling that this code is writing out chunks no matter what this flag says, so the naming is too close.

maybe it's more like a 'progressive write' kind of thing?

Reply:
I'll change it to progressiveWrite then.